README.md

```diff
@@ -42,6 +42,38 @@ Idiomatically the name is last element of the path but it need not be.
 For instance, the last element of the path might be the package's version, and the package would benefit
 from a more descriptive name.
 
+Docker Wrapper
+--------------
+The generator can also be executed via a Docker container. The image containes `protoc`, the microgenerator
+binary, and the standard API protos.
+
+```bash
+$ docker pull gcr.io/gapic-images/gapic-generator-go
+$ docker run \
+  --rm \
+  --user $UID \
+  --mount type=bind,source=/abs/path/to/protos,destination=/in/protos,readonly \
+  --mount type=bind,source=$GOPATH/src,destination=/out/ \
+  --env "GO_GAPIC_OPT=github.com/package/import/path;name" \
```

Comment:
I do not think it should hold up this PR, but we should actually make a convention for passing options to protoc. Since there is only one option switch available per plugin, it might be wise to use URI query string syntax (`key=value&key2=value2`) for extensibility. This should be something that has easy tooling to consume in every language. Thoughts?

Summoning @andreamlin who might also be interested in this.

Reply:
`only one option switch available per plugin` - but it can be present several times in a single invocation of `protoc`. Not sure we need the `&` part. imo it would be more user-friendly to supply multiple instances of the option with their `key=value` pairs.

```bash
protoc ... \
  --go_gapic_out=/out/ \
  --go_gapic_opt=key1=value1 \
  --go_gapic_opt=key2=value2 \
  my.proto
```

But as we wrap the plugin invocation more, this becomes more of an implementation detail and less of a UX concern. Thus, I'm not too concerned on how we do this

Reply:
Oh, I see. What happens if `--go_gapic_out` is specified multiple times? I just checked, and `string parameter` in `plugin.proto` [is not a repeated field](https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/compiler/plugin.proto#L74), so what does the plugin get if it is multiply specified?

Reply:
They are concatenated with commas. The same example input is presented to the plugin impl as
`key1=value1,key2=value2,key3=value3`.

The `gcli` plugin requires multiple options (two to be exact) and I went with the key:value approach similar to what I've shown above.

Reply:
I couldn't find any docs detailing this behavior but here is the [source](https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/compiler/command_line_interface.cc#L1694)!

Reply:
Okay, in that case, let's comma separate, and then it does not matter which approach is used; they will both work. (It is unfortunate that we lose URL escaping tools, but oh well.)